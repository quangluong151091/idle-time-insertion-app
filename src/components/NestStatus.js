import React from 'react';
import ReactSVG from 'react-svg';
import Timeline from './timeline/timeline/containers';
import DragDropModal from './DragDropModal.js';
import Root from './timeline/Root.js';

class NestStatus extends React.Component {
    constructor(props) {
        super(props);
        this.closeModal = this.closeModal.bind(this);
        this.state = {
            item: null,
            part: null,
            showModal: false
        }
    }
    onDragOver = (ev) => {
        ev.preventDefault();
        var dropzoneElement = ev.target;

        // feedback the possibility of a drop
        if (dropzoneElement.classList.contains('dropzone'))
            dropzoneElement.classList.add('drop-target');
    }
    formatDateStr(d) {
        d = d.toString();
        d = d.split("G")[0]
    
        return d;
    }
    onDrop = (ev) => {
        ev.preventDefault();
        var data = JSON.parse(ev.dataTransfer.getData("text"));
        this.setState({ item: data.content, part: ev.target.id });
        this.setState({showModal: true});

        // remove the drop feedback style
        if (ev.target.classList.contains('dropzone')) {
            ev.target.classList.remove('drop-target');
        }
    }
    onDragLeave = (ev) => {
        ev.preventDefault();
        // remove the drop feedback style
        if (ev.target.classList.contains('dropzone')) {
            ev.target.classList.remove('drop-target');
        }
    }
    closeModal() {
        this.setState({ showModal: false });
    }
    render() {
        const path = this.props.name + '.svg';

        return (
            <td className="main">
                <h1>{this.props.name}</h1>
                <br />
                <div style={{ width: '95%', float: "left", paddingBottom: '10px' }}>
                    <ReactSVG
                        src={path}
                        evalScripts="always"
                        fallback={() => <span>Error!</span>}
                        loading={() => <span>Loading</span>}
                        onInjected={(error, svg) => {
                            if (error) {
                                console.error(error)
                                return
                            }
                        }}
                        onDragOver={(e) => this.onDragOver(e)}
                        onDrop={(e) => { this.onDrop(e) }}
                        onDragLeave={(e) => this.onDragLeave(e)}
                    />
                </div>

                <div id="timeline" style={{ clear: 'both', paddingTop: '20px' }}>
                    <Root>
                        <Timeline name={this.props.name}/>
                    </Root>
                </div>

                <DragDropModal name={this.props.name} show={this.state.showModal} 
                    closeModal={this.closeModal} item={this.state.item} part={this.state.part}/>
            </td>
        );
    }
}

export default NestStatus