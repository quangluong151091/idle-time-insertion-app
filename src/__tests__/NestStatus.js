import React from 'react';
import { shallow } from 'enzyme';

import NestStatus from '../components/NestStatus';

describe('NestStatus', () => {
  it('should render correctly', () => {

    const component = shallow(<NestStatus />);

    expect(component).toMatchSnapshot();
  });
  it('should be able to highlight dropzone when dragged over', () => {

    const component = shallow(<NestStatus />);
    const containFc = jest.fn();
    const removeFc = jest.fn();
    const addFc = jest.fn();
    component.find("ReactSVG").simulate("dragover", { preventDefault: () => { }, 
      'target': {classList: { contains: containFc.mockReturnValue(true), remove: removeFc, add: addFc }}})
  });
  it('should be able to remove styles of dropzone when leaving', () => {

    const component = shallow(<NestStatus />);
    const containFc = jest.fn();
    const removeFc = jest.fn();
    const addFc = jest.fn();
    component.find("ReactSVG").simulate("dragleave", { preventDefault: () => { }, 
      'target': {classList: { contains: containFc.mockReturnValue(true), remove: removeFc, add: addFc }}})
    
  });
  it('should be able to drop', () => {

    const component = shallow(<NestStatus />);
    const containFc = jest.fn();
    const removeFc = jest.fn();
    const addFc = jest.fn();
    component.find("ReactSVG").simulate("drop", { preventDefault: () => { }, 
      'target': {classList: { contains: containFc, remove: removeFc, add: addFc }, id: "29484"},
      dataTransfer: {getData: jest.fn().mockReturnValue("[{}]")}});
    expect(component.state("showModal")).toBe(true);
  });
});