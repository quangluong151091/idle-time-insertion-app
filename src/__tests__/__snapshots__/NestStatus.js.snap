// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestStatus should render correctly 1`] = `
<td
  className="main"
>
  <h1 />
  <br />
  <div
    style={
      Object {
        "float": "left",
        "paddingBottom": "10px",
        "width": "95%",
      }
    }
  >
    <ReactSVG
      evalScripts="always"
      fallback={[Function]}
      loading={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onInjected={[Function]}
      renumerateIRIElements={true}
      src="undefined.svg"
      svgClassName={null}
      svgStyle={Object {}}
      wrapper="div"
    />
  </div>
  <div
    id="timeline"
    style={
      Object {
        "clear": "both",
        "paddingTop": "20px",
      }
    }
  >
    <_default>
      <Connect(timelineLayout) />
    </_default>
  </div>
  <DragDropModal
    closeModal={[Function]}
    item={null}
    part={null}
    show={false}
  />
</td>
`;
